### ISSUE TITLE
BUG: Connectivity Events not reliable?

### ISSUE THREAD

**Issue by netdata-be** [teslamotors/fleet-telemetry#244]
Hi, 

When using the command-api I see my vehicle is going to sleep and is offline.

```
/api/1/vehicles/<VIN>  | jq .response.state
"offline"
```

I have enabled the connectivity to log to stdout, there I find these events:

```
{"ConnectionID":"f6433181-31d7-4831-aff3-78979bb019f2","CreatedAt":1732880726,"Status":"CONNECTED","Vin":"xxx"}
{"ConnectionID":"f6433181-31d7-4831-aff3-78979bb019f2","CreatedAt":1732880726,"Status":"DISCONNECTED","Vin":"xxx"}
{"ConnectionID":"0042e097-2122-414b-8dcf-417b5d41b200","CreatedAt":1732880726,"Status":"CONNECTED","Vin":"xxx"}
{"ConnectionID":"63b9e371-7d2b-4277-9b8e-96171a847b42","CreatedAt":1732881637,"Status":"CONNECTED","Vin":"xxx"}
```

A few questions here:
1. Why does the vehicle connect + reconnect at the same time?
2) I see 2 CONNECTED events, but none DISCONNECTED event
3) Why does the vehicle have 2 CONNECTED Events ?

**Comment by Bre77** [comment]
I've also found these events to be unreliable. I was really looking forward to having a strong realtime awake vs asleep/offline signal, but am seeing similar behaviour from my Model 3.

![Screenshot_20241130-070219.png](https://github.com/user-attachments/assets/963dd983-2bfd-4751-a478-9aee2200d11d)

**Comment by stx** [comment]
Yeah, looking at the times when data is sent by Fleet Telemetry (along with the car timestamps of those data points), and looking at when Fleet Telemetry reports Connected/Disconnected, there is basically no correlation.

While "Disconnected", data is still sent and the car is still awake and active. Happens across all models. There's no way this can be used for sleep/wake detection as is. Might need another solution.

**Comment by a2105z** [comment]
Still open and unresolved as of Aug 2026 — last maintainer activity on this issue was none; community reports date back to Nov 2024.

Root cause on the server: connectivity was emitted per WebSocket, so overlapping sockets (wifi + cellular / reconnect races) produced false `DISCONNECTED` while another socket was still streaming.

Proposed fix (VIN-level CONNECTED on first socket / DISCONNECTED on last): https://github.com/teslamotors/fleet-telemetry/pull/540

cc @patrickdemers6 @agbpatro @aaronpkahn @ThomasAlxDmy — this has been a `priority:high` bug for a long time with no acknowledgment. Please review when you have a chance.
